scripts/rebuildstaging.py

```diff
@@ -184,9 +184,8 @@ def rebuild_staging(config, print_details=True, push=True):
     merge_conflicts = []
     not_found = []
     all_configs = list(config.span_configs())
-    with ExitStack() as stack:
-        for path, config in all_configs:
-            stack.enter_context(OriginalBranch(get_git(path)))
+    for path, config in all_configs:
+        with OriginalBranch(get_git(path)):
```

Comment:
The `if` block at

https://github.com/dimagi/commcare-hq/blob/fcc9d7eb09fb15d0819e54aba47737743c6990ca/scripts/rebuildstaging.py#L248-L256

Used to be in the `with` block, but now is in the `for` loop. I think that's not what you wanted. Also, this is a pretty long function, which is probably why that happened without your noticing...

Reply:
good find, thanks.  I was hoping to avoid getting sucked in to refactoring this, but I think this proves that that's necessary